python/bifrost/units.py

```diff
@@ -31,13 +31,16 @@
 ureg = pint.UnitRegistry()
 
 def convert_units(value, old_units, new_units):
-	old_quantity = value * ureg.parse_expression(old_units)
-	try:
-		new_quantity = old_quantity.to(new_units)
-	except pint.DimensionalityError:
-		raise ValueError("Cannot convert units %s to %s" %
+    old_quantity = value * ureg.parse_expression(old_units)
+    try:
+        new_quantity = old_quantity.to(new_units)
+    except pint.DimensionalityError:
+        raise ValueError("Cannot convert units %s to %s" %
 		                 (old_units, new_units))
-	return new_quantity.magnitude
+    if value >= 0:
+        return new_quantity.magnitude
+    else:
+        return -1 * new_quantity.magnitude
```

Comment:
My interpretation is that you need to reverse the fine_time axis before merging, and then probably again after re-splitting. There could be something else going on though. We could try plotting the data with/without the reversal(s).

The space problem is a bit tricky. At some point I want implement a fused fft+fftshift+detect, which will help.